### ISSUE TITLE
How to get log file and export output to pdf file?

### ISSUE THREAD

**Issue by greenwind1202** [gali8/Tesseract-OCR-iOS#172]
I'm trying to create demo iOS app using Tesseract-OCR-iOS library. I want to get log file and write output to pdf file. After investigating, I set some parameter into config file as below: 
debug_file                         tesseract.log
tessedit_create_pdf          T

But I can not find log file and *.pdf file. Where am I wrong?

Thanks and best regards.

**Comment by ws233** [comment]
I might be wrong, but an issue might occur due to the forbidden write permissions in the app root directory. Can you prove that tess doesn't try to create those files in the current folder? Can you also try any absolute full paths to any writable folders?

**Comment by ws233** [comment]
Exactly! I did try an absolute path to Documents folder, and a log file has appeared in it.

**Comment by greenwind1202** [comment]
Thanks for your quick reply but can you get more detail? How to set absolute path? Like as below?
debug_file absolutepath

**Comment by ws233** [comment]
I put the following line to the config file:
debug_file /Users/yourUser/Documents/tesseract.log
Also I did try the code in simulator.
The log is not empty. But you should at least initialize a tesseract and perhaps recognize any image.

I guess that 'debug_file' is a init time variable. So it's possible to set it only from the config file, or config dictionary while call to init... Don't use 'setVariableValue:forKey' for that purpose. It won't work.

**Comment by greenwind1202** [comment]
Thank you so much. I can get log file without problem. I have another question about generate output to pdf file. Currently, does library support to do this?

**Comment by ws233** [comment]
Yes, it is. But as I've already told the default output for pdf is the executable folder, which is write protected. You should try to change output dir, if it's possible with some config variables. If it's not, you might take a look on upstream tesseract sources. TessPDFRenderer could be a starting point for your investigations. If you succeed pls don't forget to submit a pull request, so we all are able to use this functionality.

**Comment by greenwind1202** [comment]
Yes.. I got it. That's exactly what I thought. :).   Again, thank you for your support.

**Comment by ws233** [comment]
You're welcome. Feel free to ask any questions. We'll try to provide with so many support as we can.

**Comment by ws233** [comment]
Finally, I've written a code, which generates pdf NSData, which could be saved to Documents folder, for example. Inspite of tesseract generates an output, it's corrupted.
So I'm trying to understand what's wrong with my code.

**Comment by ws233** [comment]
I've succeeded with this and with PDF output. I'll upload it soon.

**Comment by ws233** [comment]
Pls, refer to #186.

The following things should be mentioned regarding the patch:
* Leptonica has been built with libz, libpng, libjpeg and liftiff.
* Building script has been changed for that.
https://github.com/ashtons/libtiff-ios repository is used to build dependent libraries. The whole repository should be downloaded into 'TesseractOCR/libtiff-ios-master' folder. And all 3 .sh scripts should be run before running the build_dependencies.sh script.
* Podfile.lock has been updated to support CocoaPods 0.36.4

There are still few issues with PDF output:

* For example, I couldn't produce PNG output for the original well_scaned_page.png file. The reason possibly is in the PDF renderer when given Truecolor PNG input files. So I've changed the format of the file from PNG to JPEG and it works fine. Refer tests for details.
* Most recent Tesseract code related to PDF generation has been changed significantly. So it'll take some efforts to update to a new upstream tess code. Also, I've faced a few linker issues, while trying this. Perhaps, I'll need some help for this.
